android/src/main/java/app/notifee/core/NotificationReceiverHandler.java

```diff
@@ -0,0 +1,136 @@
+package app.notifee.core;
+
+/*
+ * Copyright (c) 2016-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import static app.notifee.core.NotificationPendingIntent.EVENT_TYPE_INTENT_KEY;
+import static app.notifee.core.NotificationPendingIntent.NOTIFICATION_ID_INTENT_KEY;
+import static app.notifee.core.event.NotificationEvent.TYPE_ACTION_PRESS;
+import static app.notifee.core.event.NotificationEvent.TYPE_PRESS;
+
+import android.app.RemoteInput;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import androidx.core.app.NotificationManagerCompat;
+import app.notifee.core.event.InitialNotificationEvent;
+import app.notifee.core.event.MainComponentEvent;
+import app.notifee.core.event.NotificationEvent;
+import app.notifee.core.model.NotificationAndroidPressActionModel;
+import app.notifee.core.model.NotificationModel;
+
+public class NotificationReceiverHandler {
+  public static final String REMOTE_INPUT_RECEIVER_KEY =
+      "app.notifee.core.ReceiverService.REMOTE_INPUT_RECEIVER_KEY";
+  private static final String TAG = "NotificationReceiverHandler";
+
+  static void handleNotification(Context context, Intent intent) {
```

Comment:
a general thought for the whole class - can intent ever be null? Would be interesting to put NotNull annotations on there and see if static analysis from compiler complains about it from earlier in the call chain or not

Reply:
I believe it can't but that's a good idea. 💯